bin/testEnvLocalK8S.sh

```diff
@@ -111,6 +127,8 @@ function startLocalApiserver() {
         --port 8080 -v 2 --insecure-bind-address 0.0.0.0 \
         > ${LOG_DIR}/apiserver.log 2>&1 &
     echo $! > $LOG_DIR/apiserver.pid
+    # make sure apiserver is actually alive
+    kill -0 $(cat $LOG_DIR/apiserver.pid)
```

Comment:
never heard of the 0 signal before - what is kill -0 ?
edit: found https://unix.stackexchange.com/questions/169898/what-does-kill-0-do
which isn't documented in mac man page for kill but does appear to work - cool!

Reply:
In my earlier testing etcd failed to launch on my Mac since the script downloaded a Linux binary.  This script blindly continued (being ignorant of the failure) so I wanted to add something to make the failure be caught and more evident.  If the "&" suffix wasn't present then the failure-to-launch is caught by "pipefail", but otherwise the error of the child process isn't caught.  I consulted various options and kill -0 seems to cover posix (other options for querying the pid seemed to be Linux-specific).  I'm option to other ways to catch this error.  For now the error is going to be typical/expected on Mac given the absence of apiserver (though on 2nd thought the script will likely fail earlier over the inability to download apiserver).